### PR TITLE
Auth logic

### PR DESCRIPTION
**Pull request by AnuragSahu11**
*Added auth login
*Added Redux and redux states
*Added basic firestore calls

### CODE REVIEW

File: src/firebase/firestore-methods.jsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    batch.set(userRef, {
+      firstName,
+      lastName,
+      email,
+      bio: "",
+      dp: "",
+      website: "",
+    });
+    const postRef = doc(db, userID, "posts");
+    batch.set(postRef, { posts: [] });
```
Comment (shivampandey0) [inline src/firebase/firestore-methods.jsx:32]: Nice to see the use of firebase's method here to write multiple data in a single query

**Review by shivampandey0 — APPROVED** [review]
Great folder structure and code structuring to keep all the network methods in one place and setup of Redux beforehand.

Learned a lot from this code.

Good to merge

**Review by GuptaEesh — COMMENTED** [review]
- Implementation of redux looks good.
- Nice to see use of firebase for the project
- good to merge

File: src/firebase/firestore-methods.jsx (hunk 1 above)
Comment (AnuragSahu11) [reply src/firebase/firestore-methods.jsx:32]: Thanks

**Pull request merged**